src/autopas/selectors/tuningStrategy/BayesianClusterSearch.h

```diff
@@ -166,9 +167,14 @@ class BayesianClusterSearch : public TuningStrategyInterface {
     _currentIteration = iteration;
     ++_currentNumEvidence;
     _currentAcquisitions.clear();
+
+    _traversalTimes[_currentConfig] = time;
   }
 
+  inline long getEvidence(Configuration configuration) const override { return _traversalTimes.at(configuration); }
+
   inline void reset(size_t iteration) override {
+    _gaussianCluster.clear();
```

Comment:
This is intended. GaussianCluster needs a bunch of evidence to work properly. We keep the evidence over several tuning phases to avoid constant resampling.
```suggestion
```

Reply:
when do you actually clear this?

Reply:
Currently we only decay their influence over time. Theoretically we could delete them if their influence becomes negligible. But we have not yet encountered any problems in our tests yet. To cause performance problems you probably have to set maxEvidence very high.

However, if this is a problem, the best option would probably be to restart the whole strategy after a few tuning phases.

```suggestion
    if (++phasesSinceLastReset >= phasesPerReset) {
        _gaussianCluster.clear();
        _firstTuningPhase = true;
        phasesSinceLastReset = 0;
    }
```

Reply:
On that note you probably need to add the following after resetting fullSearch.

`_currentConfig = _fullSearch.getCurrentConfiguration();`

https://github.com/AutoPas/AutoPas/blob/b2f476522ed55acd098edffd951066d581e918c1/src/autopas/selectors/tuningStrategy/BayesianClusterSearch.h#L188-L190

Reply:
@Njan0 do I understand it correctly that the last two suggestions are not really necessary?
I just assumed that the gaussianCluster should also be reset, but if you say there are no problems by not calling clear() than I don't see any either.
Is your last comment also related to this or is it a general suggestion?

Reply:
My last comment would generally be the correct implementation. But this is only a problem if someone calls reset() although tune() still returns true. So something that currently should never happen. But maybe that will change one day. ;)